### PR TITLE
Advanced search in Teambuilder

### PR DESCRIPTION
**Pull request by littlefoot22**
Really like this project and want to contribute so I took a shot at one of the items on the new develops issue. 

Please let me know if this is on the right track / feedback / what to add / is this still needed?

Supports search on / Pokémon / ability / item / move / anything in unpacked team:

screen shot 2018-07-12 at 8 51 53 pm
screen shot 2018-07-12 at 8 51 57 pm
screen shot 2018-07-12 at 8 52 08 pm
screen shot 2018-07-12 at 8 52 11 pm
screen shot 2018-07-12 at 8 58 20 pm
screen shot 2018-07-12 at 9 01 17 pm
screen shot 2018-07-12 at 9 05 53 pm
screen shot 2018-07-12 at 9 07 14 pm

### CODE REVIEW

File: js/client-teambuilder.js [hunk 1]
```diff
@@ -2570,6 +2584,13 @@
 			}
 			this.chartSet(val, selectNext);
 		},
+		searchChange: function (e) {
+			//91 for right CMD / 93 for left / CMD 17 for CTL
+			if (e.keyCode != 91 && e.keyCode != 93 && e.keyCode != 17) {
```
Comment (HoeenCoder) [inline js/client-teambuilder.js]: Can use `!==` here as (iirc) the type will always be `number`

Comment (littlefoot22) [reply js/client-teambuilder.js]: Thanks for the feedback! I'll update the PR when I'm home

**Comment by Zarel** [comment]
;_; I keep on having these refactors being submitted when I'm in the middle of rewriting the client in React...

**Comment by Zarel** [comment]
I actually love this but also merging it in is going to be hell...

**Comment by littlefoot22** [comment]
I read about the refactor in the comments, I mostly did this to get a familiar with the project. I won’t be offended if you dont merge this lol. I am more excited to work on the new client when it arrives :D

**Comment by scheibo** [comment]
> I actually love this but also merging it in is going to be hell...

This looks pretty mergeable.

**Comment by Zarel** [comment]
In hindsight, yeah. There's no reason not to merge.

**Comment by scheibo** [comment]
@littlefoot22 - do you have a Smogon username that you'd like to have tagged when I post the [user-visible changelog](https://gist.github.com/scheibo/95a34d58b842d9639c06a35e970f18aa) to the [update thread](https://www.smogon.com/forums/threads/pok%C3%A9mon-showdown-updates.3650958/) after the next server restart?

**Comment by littlefoot22** [comment]
> @littlefoot22 - do you have a Smogon username that you'd like to have tagged when I post the [user-visible changelog](https://gist.github.com/scheibo/95a34d58b842d9639c06a35e970f18aa) to the [update thread](https://www.smogon.com/forums/threads/pok%C3%A9mon-showdown-updates.3650958/) after the next server restart?

@scheibo 
my username of the forums is buck.20.two and on showdown it is littlefoot22

**Pull request merged**